{
  if (backupOp != null) {
    try {
      backupOp.onExecutionFailure(e);
    }
 catch (    Throwable t) {
      getLogger().warning("While calling operation.onFailure(). op: " + backupOp,t);
    }
  }
}
